src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -793,24 +790,23 @@ ov::Tensor ov::npuw::util::concat(const std::vector<ov::Tensor>& tt, std::size_t
         uint8_t* pDst = static_cast<uint8_t*>(tnew.data());
 
         const bool is_4bit = (type == ov::element::i4 || type == ov::element::u4);
-        for (std::size_t t_idx = 0; t_idx < tt.size(); t_idx++) {
+        ov::parallel_for(tt.size(), [&](size_t t_idx) {
             const uint8_t* pSrc = static_cast<const uint8_t*>(tt[t_idx].data());
 
             const auto copy_size = lens[t_idx] * shape[1] * shape[2];
             const auto copy_len = is_4bit ? copy_size / 2 : copy_size * type.size();
 
             std::copy_n(pSrc, copy_len, pDst);
             pDst += copy_len;
```

Comment:
in case of parralel_for this is not valid - you need to use t_idx as a reference and compute pDst every time, not increment it

Reply:
Nice catch.
Why we didn't catch it with tests?

Reply:
Fixed it it latest [commit](https://github.com/openvinotoolkit/openvino/pull/30899/commits/28e7cb6f3236f916fd11c0072a07a46edd8707e8).